src/illuminatio/illuminatio_runner.py

```diff
@@ -2,25 +2,29 @@
 This file contains the implementation of the illuminatio runner which
 actively executes network policy tests inside the kubernetes cluster itself
 """
+import ipaddress
 import json
 import logging
 import os
 import socket
 import subprocess
-import tempfile
 import time
-from xml.etree import ElementTree
+import platform
 import yaml
+import nmap
 
 import click
 import click_log
 import docker
 import kubernetes as k8s
-from nsenter import Namespace
 
 from illuminatio.host import Host, ConcreteClusterHost
 from illuminatio.k8s_util import create_test_output_config_map_manifest
 
+# Otherwise we get an error on Mac
+if platform.system() == "Linux":
+    from nsenter import Namespace
+
```

Comment:
What happens when the code below using that import is executed? If you didn't execute that code, I would prefer importing this in the function where it is used (maybe remove the plattform check, keep the comment)

Reply:
It will explode :) Just joking it will run in an error but this is fine since we only need this check for the test setup (we don't support the runner on MacOS or Windows or some other OS). I wouldn't prefer importing in the function, this will probably a performance penalty and I don't see how this would be better than the current approach.